### PR TITLE
added PEP-343 conformance (Closes blue-yonder/turbodbc#149)

### PR DESCRIPTION
**Pull request by AtomBaf**

### CODE REVIEW

**Comment by MathMagique** [comment]
Hi! Thanks for the contribution! Could you please add unit tests for the functionality you added? I'd suggest to use a connection in a context manager successfully, and then assert that using the connection outside of the context manager raises an exception. You could use https://github.com/blue-yonder/turbodbc/blob/master/python/turbodbc_test/test_connection.py#L10 as a template for a new test.
Similarly, a test for the cursor context could be put in https://github.com/blue-yonder/turbodbc/blob/master/python/turbodbc_test/test_cursor_basics.py#L22.

**Comment by AtomBaf** [comment]
Here are the tests. Hopefully it will be easy to merge.

File: python/turbodbc_test/test_connection.py [hunk 1]
```diff
@@ -151,3 +151,13 @@ def test_autocommit_querying(dsn, configuration):
     connection.autocommit = True
     assert connection.autocommit == True
     connection.close()
+
+@for_one_database
+def test_pep343_with_statement(dsn, configuration):
+
+    with connect(dsn, **get_credentials(configuration)) connection:
```
Comment (MathMagique) [inline python/turbodbc_test/test_connection.py]: This `with` statement is missing an `as`, that's why the build breaks

**Comment by MathMagique** [comment]
I just made the necessary change.

**Comment by MathMagique** [comment]
Thanks for your contribution!

**Pull request merged**